config/routes.rb

```diff
@@ -49,8 +49,9 @@
         resources :field_update_comments, only: %w[index create]
       end
 
-      scope '/flagged-applications' do
-        put 'update' => 'flagged_applications#update'
+      resources :flagged_applications, path: 'flagged-applications', only: %w[index update]
+      scope 'flagged-applications' do
+        get 'record-set/:id' => 'flagged_applications#record_set'
```

Comment:
**note:** we already have talks to move and redesign our APIs. For the url `/api/v1/flagged-applications?type=pending` I would think it returns a list of applications that have been flagged but instead it is returning a list of listings that have flagged applications. Same happens with the web page url `/applications/flagged?type=pending`. Maybe something like `/listings/applications?type=flagged` might

Reply:
Yes, I agree that the naming conventions in the code is inconsistent with the data being processed, and it was a big source of confusion for me as a relative newcomer to the LAP repository. [This](https://github.com/SFDigitalServices/sf-dahlia-lap/pull/24#issuecomment-389608615) is when the naming got muddled for the sake of "clean code."

Under the hood, we are using SOQL queries to load data from the `Flagged_Record_Set__c` Salesforce table, so it would make more sense to organize the endpoints like:
```ruby
# used to update status and comment of a single flagged application
resources :flagged_applications, path: 'flagged-applications', only: %w[update]
# used to get pending flagged record sets, marked-duplicate flagged record sets, and individual flagged record sets
resources :flagged_application_record_sets, path: 'flagged-application-record-sets', only: %w[index show]
```

This isn't in the scope of the ticket, but happy to create a new ticket to capture this.

Reply:
I think its fine for now since we will be creating ticket in the future for new APIs, just wanted to note. Thanks!